ultra/ultra/evaluate.py

```diff
@@ -308,41 +340,40 @@ def evaluate_saved_models(
         + "-evaluation"
     )
 
-    ray.init()
-    try:
-        for episode in episodes(
-            number_of_checkpoints,
-            etag=etag,
-            log_dir=log_dir,
-        ):
-            # Obtain a checkpoint directory for each agent.
-            current_checkpoint_directories = {
-                agent_id: agent_directories[episode.index]
-                for agent_id, agent_directories in agent_checkpoint_directories.items()
-            }
-            episode.eval_mode()
-            episode.info[episode.active_tag] = ray.get(
-                [
-                    evaluate.remote(
-                        experiment_dir=experiment_dir,
-                        agent_ids=agent_ids,
-                        policy_classes=policy_classes,
-                        seed=episode.eval_count,
-                        checkpoint_dirs=current_checkpoint_directories,
-                        scenario_info=scenario_info,
-                        num_episodes=num_episodes,
-                        max_episode_steps=max_episode_steps,
-                        timestep_sec=timestep,
-                        headless=headless,
-                        log_dir=log_dir,
-                    )
-                ]
-            )[0]
-            episode.record_tensorboard(recording_step=episode.index)
-            episode.eval_count += 1
-    finally:
-        time.sleep(1)
-        ray.shutdown()
+    for agent_id, checkpoint_directories in agent_checkpoint_directories.items():
+        num_of_checkpoints = len(checkpoint_directories)
+        ray.init()
+        try:
```

Comment:
Just curious, why was the ray initialization and `try` statement moved inside the loop?

Reply:
Discussed offline